src/containerapp/azext_containerapp/containerapp_decorator.py

```diff
@@ -797,6 +813,7 @@ def parent_construct_payload(self):
         config_def = deepcopy(ConfigurationModel)
         config_def["secrets"] = secrets_def
         config_def["activeRevisionsMode"] = self.get_argument_revisions_mode()
+        config_def["targetLabel"] = self.get_argument_target_label()
```

Comment:
The targetLabel should only be set when the revision mode is `Labels`, right?
We can add argument validation for this case.

Reply:
I will add this for create, but it's harder for up and update. revisions-mode and target-label are both saved in the app model and can be updated independently. target-label is also not cleared from the model if you leave labels mode, so I don't want to generate errors based on its presence.

The RP will enforce that target-label is required in labels mode. It will not complain if target-label is present outside of labels mode.